File: go/hamming/hamming.go
// Package hamming implements routines to calculate the
// Hamming Distance between two DNA strands.
package hamming

import "fmt"

// Distance returns the Hamming Distance bewteen
// the provided string inputs that represent DNA strands.
// Only strands of equal length can be processed.
func Distance(a, b string) (int, error) {
	ar, br := []rune(a), []rune(b)
	if len(ar) != len(br) {
		return 0, fmt.Errorf("The length of the DNA strands are not equal: %d != %d",
			len(ar), len(br))
	}
	distance := 0
	for i := range ar {
		if ar[i] != br[i] {
			distance++
		}
	}

	return distance, nil
}

File: python/hamming/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    """
    Calculate the Hamming Distance between two DNA strands.
    """
    if len(strand_a) != len(strand_b):
        raise ValueError('DNA strand lengths do not match.')
    pairs = zip(strand_a.lower(), strand_b.lower())
    return sum([ch1 != ch2 for ch1, ch2 in pairs])
